### PR TITLE
adds automaxprocs and automemlimit

### PR DESCRIPTION
**Pull request by vroldanbet**
closes https://github.com/authzed/spicedb/issues/498

so that SpiceDB becomes aware of cgroup cpu and
memory limits, and automatically
adjusts the go runtime gomaxprocs and gomemlimit
parameters

### CODE REVIEW

**Comment by vroldanbet** [comment]
I have read the CLA Document and I hereby sign the CLA

File: cmd/spicedb/main.go [hunk 1]
```diff
@@ -3,15 +3,18 @@ package main
...
+	"github.com/KimMachineGun/automemlimit/memlimit"
 	mcobra "github.com/muesli/mango-cobra"
 	"github.com/muesli/roff"
 	"github.com/rs/zerolog"
+	slogzerolog "github.com/samber/slog-zerolog/v2"
 	"github.com/sercand/kuberesolver/v5"
 	"github.com/spf13/cobra"
+	"go.uber.org/automaxprocs/maxprocs"
 	"google.golang.org/grpc/balancer"
-	_ "google.golang.org/grpc/xds"
```
Comment (ecordell) [inline cmd/spicedb/main.go:14]: Return this please! This is what enables `xds` support for the hashring.

(We don't have tests for it which is why nothing failed - we should though)

**Review by ecordell — APPROVED** [review]
https://github.com/authzed/spicedb/issues/498#issuecomment-1247036753

^ this comment still stands, but I think for the most part it's not a huge deal because you can always set GOMAXPROCs explicitly, or you can roll the pod to pick up the new available values. It might be worth documenting this behavior though, since I don't think there are any kube docs for it.

File: cmd/spicedb/main.go (hunk 1 above)
Comment (vroldanbet) [reply cmd/spicedb/main.go:14]: shoot, sorry!

**Review by ecordell — APPROVED** [review]

**Pull request merged**